applications/serial_lte_modem/src/slm_at_host.c

```diff
@@ -317,6 +333,81 @@ static int handle_at_slmuart(const char *at_cmd, uint32_t *baudrate)
 	return ret;
 }
 
+void silence_timer_handler(struct k_timer *timer)
+{
+	ARG_UNUSED(timer);
+
+	/* subordinate silence time-out */
+	exit_datamode();
+	if (slm_tcp_get_datamode()) {
+		slm_tcp_set_datamode_off();
+	}
+	if (slm_udp_get_datamode()) {
+		slm_udp_set_datamode_off();
+	}
+	datamode_off_pending = false;
+}
+
+K_TIMER_DEFINE(silence_timer, silence_timer_handler, NULL);
+
+static void raw_send(struct k_work *work)
+{
+	const char *terminator = CONFIG_SLM_DATAMODE_TERMINATOR;
+	int len = strlen(terminator);
+	int64_t silence_limit = CONFIG_SLM_DATAMODE_SILENCE * MSEC_PER_SEC;
+	int64_t silence_delta = k_uptime_delta(&silence_start);
+	int err;
+
+	ARG_UNUSED(work);
+
+	LOG_HEXDUMP_DBG(at_buf, at_buf_len, "RX");
+
+	/* check leading silence */
+	if (silence_delta < silence_limit) {
```

Comment:
I'm not sure if I understand the need for `silence_start/delta/limit` logic. Can't we just start the timer when the terminating sequence is detected and then just check it's status in this function (there are functions like `k_timer_remaining_get()` for instance). 

If there's new data received when the timer is still running, it can be stopped and the terminating sequence injected into the data, otherwise the timeout handler will kick in finishing the datamode. 

Plus the terminating sequence should be verified in any case I believe, what if `+++` was received before, and then before the silence timer runs out another `+++` sequence is received. The former one shall be treatead as a data then but we stil shall restart the timer (as new termination sequence was received). At least that's how I understood the algorithm from the wiki.

Reply:
There is a need of leading minimal 1s second of UART silence. The _if_ here is to check whether any data is received within 1s interval from last UART RX. If so and already received "+++", we need to cancel the 1s timer used to check subordinator UART silence (also 1s), and handle received data as normal raw data.

We only need one timer to check the subordinator UART silence, which is started right after exact "+++" is received.